modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/index.d.ts

```diff
@@ -0,0 +1,227 @@
+	 */
+	after_: string[];
+
+	/**
+	 * Array of listeners that will invoke before the displaced function.
+	 */
+	before_: string[];
+
+	/**
+	 * The name of the displaced function.
+	 */
+	fnName_: string;
+
+	/**
+	 * The displaced function.
+	 */
+	fn_: () => void;
+
+	/**
+	 * The object hosting the method to displace.
+	 */
+	obj_: object;
+
+	/**
+	 * Creates handle for detaching listener from displaced function.
+	 */
+	createHandle(fn: () => void, before: boolean): object;
+
+	/**
+	 * Detaches listener from displaced function.
+	 */
+	detach_(fn: () => void, before: boolean): void;
+
+	exec(...args: any[]): any;
+
+	/**
+	 * Registers an AOP listener.
+	 */
+	register(fn: () => void, before: boolean): EventHandle;
+
+	/**
+	 * Executes the supplied method after the specified function.
+	 */
+	static after(fn: () => void, object: object, fnName: string): EventHandle;
+
+	/**
+	 * Return an alterReturn object when you want to change the result returned
+	 * from the core method to the caller.
+	 */
+	static alterReturn(value: any): {type: string; value: any};
+
+	/**
+	 * Executes the supplied method before the specified function.
+	 */
+	static before(fn: () => void, object: object, fnName: string): EventHandle;
+
+	/**
+	 * Return a halt object when you want to terminate the execution
+	 * of all subsequent subscribers as well as the wrapped method
+	 * if it has not executed yet.
+	 */
+	static halt(value: any): {type: string; value: any};
+
+	/**
+	 * Executes the supplied method before or after the specified function.
+	 */
+	static inject(
+		before: boolean,
+		fn: () => void,
+		object: object,
+		fnName: string
+	): EventHandle;
+
+	/**
+	 * Returns object which instructs `exec` method to modify the return
+	 * value or prevent default behavior of wrapped function.
+	 */
+	static modify_(type: string, value: any): {type: string; value: any};
+
+	/**
+	 * Return a prevent object when you want to prevent the wrapped function
+	 * from executing, but want the remaining listeners to execute.
+	 * @return {Object}
+	 */
+	static prevent(): {type: string; value: any};
+}
+
+export class AutoSize {
+	constructor(inputElement: HTMLElement);
+
+	createTemplate(computedStyle: {
+		fontFamily: string;
+		fontSize: string;
+		fontStyle: string;
+		fontWeight: string;
+		letterSpacing: string;
+		lineHeight: string;
+		textTransform: string;
+	}): HTMLPreElement;
+
+	handleInput(event: string): void;
+
+	_resizeInput(inputElement: HTMLElement): void;
+}
+
+export class Disposable {
+
+	/**
+	 * Disposes of this instance's object references. Calls `disposeInternal`.
+	 */
+	dispose(): void;
+
+	/**
+	 * Checks if this instance has already been disposed.
+	 */
+	isDisposed(): boolean;
+}
+
+export class EventEmitter {
+
+	/**
+	 * Holds event listeners scoped by event type.
+	 */
+	_events: {
+		[key: string]: Array<() => void>; // @bryceosterhaus: not sure how correct this one is
```

Comment:
Oh my bad, I didn't realize you `@mentioned` me within the code itself. GH doesn't give notifications for this, it would probably be best next time to add your own code review and add comments and mention via gh, that way I will get notified and we can track the conversation a bit better that way.

I will look at these today though!